examples/v8-react18/package.json

```diff
@@ -0,0 +1,46 @@
+{
+  "name": "v8-react18",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "react-scripts": "^5.0.0"
+  },
+  "scripts": {
+    "clear": "rimraf __screenshots__",
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test --setupFiles ./setupFile.js",
+    "eject": "react-scripts eject",
+    "storybook": "storybook dev -p 9009",
+    "build-storybook": "storybook build",
+    "prestorybook": "../../scripts/e2e-prestorybook.js .",
+    "storycap:all": "yarn prestorybook && yarn test && storycap --verbose --server-timeout 100000 --server-cmd \"storybook dev -p 9009\" \"http://localhost:9009\""
+  },
+  "eslintConfig": {
+    "extends": "react-app",
+    "rules": {
+      "import/no-anonymous-default-export": "off"
+    }
+  },
+  "browserslist": [
+    ">0.2%",
+    "not dead",
+    "not ie <= 11",
+    "not op_mini all"
+  ],
+  "devDependencies": {
+    "@storybook/addon-actions": "^8.0.0",
+    "@storybook/addon-essentials": "^8.0.0",
+    "@storybook/addon-links": "^8.0.0",
+    "@storybook/addons": "^7.6.17",
```

Comment:
Is this version intentional ?

```
    "@storybook/addons": "^8.0.0",
```

I think the following error is related to this version mismatching 🤔 

https://github.com/reg-viz/storycap/actions/runs/8294518431/job/22703172139?pr=855#step:7:88

Reply:
The `@storybook/addons` package doesn't exits for Storybook@8 it has been replaced by `@storybook/manager-api` and `@storybook/preview-api`

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#new-addons-api

Reply:
Thanks! Yes, the `@storybook/addons` seems to be replaced, so we need to switch to using `@storybook/manager-api` and `@storybook/preview-api`.

However, the following files depend on `@storybook/addons`.
- packages/storycap/src/client/register.ts
- packages/storycap/src/client/with-screenshot.ts

So, I think that if we want to support the older versions of storybook, we might need to do either of the following
- A: find someway to detect the installed storybook version and maybe dynamically import the corresponding packages
- B: try to import the `@storybook/addons` package and fallback to importing `@storybook/manager-api` or `@storybook/preview-api` if it's not available

Which way seems better? B may look something like [365331c](https://github.com/reg-viz/storycap/pull/855/commits/365331cfe5e55c1dae497f8d873c8a935783a609)